pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -634,19 +665,61 @@ int kretprobe__tcp_close(struct pt_regs* ctx) {
     return 0;
 }
 
-SEC("kprobe/udp_sendmsg")
-int kprobe__udp_sendmsg(struct pt_regs* ctx) {
+SEC("kprobe/ip6_make_skb")
+int kprobe__ip6_make_skb(struct pt_regs* ctx) {
+     struct sock* sk = (struct sock*)PT_REGS_PARM1(ctx);
+     size_t size = (size_t)PT_REGS_PARM4(ctx);
+     u64 pid_tgid = bpf_get_current_pid_tgid();
+
+     size = size - sizeof(struct udphdr);
+
+     conn_tuple_t t = {};
+     if (!read_conn_tuple(&t, sk, pid_tgid, CONN_TYPE_UDP)) {
+
+         increment_telemetry_count(udp_send_missed);
+         return 0;
+     }
+
+     log_debug("kprobe/ip6_make_skb: pid_tgid: %d, size: %d\n", pid_tgid, size);
+     handle_message(&t, size, 0);
+     increment_telemetry_count(udp_send_processed);
+
+     return 0;
+}
+
+SEC("kprobe/ip_make_skb")
+int kprobe__ip_make_skb(struct pt_regs* ctx) {
     struct sock* sk = (struct sock*)PT_REGS_PARM1(ctx);
-    size_t size = (size_t)PT_REGS_PARM3(ctx);
+    size_t size = (size_t)PT_REGS_PARM5(ctx);
     u64 pid_tgid = bpf_get_current_pid_tgid();
 
+    size = size - sizeof(struct udphdr);
+
     conn_tuple_t t = {};
     if (!read_conn_tuple(&t, sk, pid_tgid, CONN_TYPE_UDP)) {
-        increment_telemetry_count(udp_send_missed);
-        return 0;
+        struct flowi4* fl4 = (struct flowi4*)PT_REGS_PARM2(ctx);
+        bpf_probe_read(&t.saddr_l, sizeof(u32), ((char*)fl4) + offset_saddr_fl4());
+        bpf_probe_read(&t.daddr_l, sizeof(u32), ((char*)fl4) + offset_daddr_fl4());
+
+        if (!t.saddr_l || !t.daddr_l) {
+            log_debug("ERR(fl4): src/dst addr not set src:%d,dst:%d\n", t.saddr_l, t.daddr_l);
+            increment_telemetry_count(udp_send_missed);
+            return 0;
+        }
+        bpf_probe_read(&t.sport, sizeof(t.sport), ((char*)fl4) + offset_sport_fl4());
+        bpf_probe_read(&t.dport, sizeof(t.dport), ((char*)fl4) + offset_dport_fl4());
+
+         if (t.sport == 0 || t.dport == 0) {
```

Comment:
nit: there is some weird indentation going on here. Can you check and make it consistent? (I hope to introduce a formatter soon).